shipyard-controller/api/service.go

```diff
@@ -130,11 +136,17 @@ func validateCreateServiceParams(params *operations.CreateServiceParams) error {
 	if params.ServiceName == nil || *params.ServiceName == "" {
 		return errors.New("Must provide a service name")
 	}
+
 	if !keptncommon.ValididateUnixDirectoryName(*params.ServiceName) {
 		return errors.New("Service name contains special character(s). " +
 			"The service name has to be a valid Unix directory name. For details see " +
 			"https://www.cyberciti.biz/faq/linuxunix-rules-for-naming-file-and-directory-names/")
 	}
+
+	if len(*params.ServiceName) > serviceNameMaxLen {
+		return errors.New("Service name need to be less than 53 characters")
+	}
```

Comment:
This check is a little bit too soft :)
Not sure if we have described this properly in #2948, but what we need is that the service name is short enough, such that the generated helm chart release name is less than 53 characters.
The helm chart release name is: `${PROJECTNAME}-${STAGENAME}-${SERVICENAME}-generated`, therefore this string needs to be less than 53 characters (e.g., `len(ServiceName) < 53 - len(ProjectName) - len(StageName) - len("generated") - 2`). 

In addition, we would like to verify this for all stages, not just the current stage (see [Keptn's Shipyard File](https://github.com/keptn/spec/blob/0.2.0-alpha/shipyard.md#stage)).

As a referene, you can take a look at the `createService` function in this file. It has the projectName and it also fetches all stage names via
```
stages, err := sm.stagesAPI.GetAllStages(projectName)
```

Reply:
thanks for your review @christian-kreuzberger-dtx 
I did not realize the stage and project. I did a small change, please let me know if that is good otherwise let's discuss the best place to add the validation.

Reply:
humm, looks like new changes in the file, will rebase and check the conflicts